src/RewardsStreamerMP.sol

```diff
@@ -503,6 +496,11 @@ contract RewardsStreamerMP is
         totalMPAccrued -= _deltaMpTotal;
         totalMaxMP -= _deltaMpMax;
         totalStaked -= amount;
+
+        // if the user can unstake it means the lock period has ended
+        // and we can reset lockUntil and totalLockTime
+        vault.lockUntil = 0;
+        vault.totalLockTime = 0;
```

Comment:
I still think we only want to reset the total lock time when all funds were unstaked.

With this, a user can:

1. Stake, lock 1 year, wait one year
2. Unstake partially, no total lock time is zero
3. Repeat

Reply:
if the user can unstake, it means the lock period ended, so the lock time is no longer need